extensions/wikibase/src/org/openrefine/wikibase/editing/MediaFileUtils.java

```diff
@@ -306,4 +370,55 @@ public MediaInfoIdValue getMid(ApiConnection connection, String siteIri) throws
             return mid;
         }
     }
+
+    /**
+     * A file read one chunk at a time.
+     */
+
+    public static class ChunkedFile {
+
+        protected FileInputStream stream;
+        protected final int chunkSize = 5000;
+        protected File path;
+        protected long bytesRead;
+
+        public ChunkedFile(File path) throws FileNotFoundException {
+            this.path = path;
+            stream = new FileInputStream(path);
+            bytesRead = 0;
+        }
+
+        /**
+         * Read the next chunk of the file.
+         *
+         * @return {File} Contains a chunk of the original file. The length in bytes is chunkSize or however much
+         *         remains of the file if the last chunk is read.
+         * @throws IOException
+         */
+        public File readChunk() throws IOException {
+            if (bytesRead >= path.length()) {
+                return null;
+            }
+
+            // Read at most the remaining bytes.
+            int bytesToRead = (int) Math.min(path.length() - bytesRead, chunkSize);
+            byte[] bytes = new byte[bytesToRead];
+            int chunkBytesRead = stream.read(bytes);
```

Comment:
With this workflow you are reading the entire chunk in memory, before writing it back to the file. I think we should be able to avoid the memory usage, no?
Intuitively, if you wrap `stream` inside a [`BoundedInputStream`](https://commons.apache.org/proper/commons-io/javadocs/api-2.4/org/apache/commons/io/input/BoundedInputStream.html) and then call [`transferTo()`](https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/io/InputStream.html#transferTo-java.io.OutputStream-) on it, it should do just that. But I haven't tried it, so it can well be that I am missing a detail there.

Reply:
I did as you suggested and it seems to work fine.

To only remaining thing is that if the BoundedInputStream should be closed according to my IDE. I've tried with both calling `closed` and adding a `try` for it, but it looks like that closes the wrapped FileInputStream; I get "java.io.IOException: Stream Closed" when calling `transferTo()` for the second chunk.

An ideas how to solve this?

Reply:
I think we just need to make sure that the `FileInputStream` gets closed once we are done, but not the `BoundedInputStream`.
I would make `ChunkedFile` implement the `Closeable` interface, delegating its `close()` method to the FileInputStream. Then it will be up to whoever creates a `ChinkedFile` to also close it afterwards (for instance with a try-with-resources block).

Reply:
After https://github.com/OpenRefine/OpenRefine/pull/6510/commits/4d5614e080ac5b6069b891ce399b14afd5f2efb6 I no longer have a warning about closing the BoundedInputStream. Do you know if I still need to do that or if it's handled by the new way of creating it?

Reply:
Yes I still think it would be worth making `ChunkedFile` implement the `Closeable` interface, so that we can properly close the `FileInputStream` that it allocates in its constructor. Otherwise, it can be that the underlying file descriptor does not get closed, which would constitute a resource leak.

Reply:
done